study.md

```diff
@@ -21,18 +21,22 @@ There is no requirement to follow any links from these pages.
 After doing all the required readings please tell us anything you were confused about.
 
 ```md
-<!-- your answer here -->
+I still find the concept of hoisting to be confusing.  Specifically, I do not understand how return does not break the function in situations involving hoisting.  The article mentions that exceptions are made due to "context", but I do not understand what this means.
```

Comment:
Here are a couple of bits of code that I wrote to demonstrate hoisting and returning.  Let me know if they're unclear.
Hoisting.
```javascript
// This will output, "[Function: sayHello]".
console.log(sayHello);

// This will work and output, "Hello".
sayHello();

// This will output, "undefined" since the declaration of "sayHi" has been
// hoisted to the top, but its definition has not been.
console.log(sayHi);

// This will throw an error for the reason above.  Since sayHi is undefined,
// it's trying to execute "undefined" as if it were a function.
sayHi();

// Function declaration.  This gets hoisted to the top.
function sayHello() {
  console.log('Hello.');
}

// Function expression.  This gets hoisted to the top, but its assignment to
// function() { ... } does not.
var sayHi = function() {
  console.log('Hi.');
};

// This will output, "[Function: sayHi]", since it's after the definition of the
// function.
console.log(sayHi);

// This will output, "Hi".
sayHi();
```
Returning.
```javascript
function a() {
  console.log('a(): Inside of a().');
  console.log('a(): Calling b() and setting its return value to bValue.');

  const bValue = b();

  console.log('a(): b() has returned.  Control has been given back to a().');
  console.log('a(): Printing bValue: ', bValue);
  console.log('a(): Returning a\'s value.');

  return 'It\'s me, a.';
}

function b() {
  console.log('b(): Inside of b().');
  console.log('b(): Calling c() and setting its return value to cValue.');

  const cValue = c();

  console.log('b(): c() has returned.  Control has been given back to b()');
  console.log('b(): Printing cValue:', cValue);
  console.log('b(): Returning B\'s value.');

  return 'It\'s me, b.';
}

function c() {
  console.log('c(): Inside of c().');
  console.log('c(): Returning c\'s value.');

  return 'It\'s me, c.';
}

console.log(
  'global: Calling a() from the global scope and setting its value to aValue.'
);

const aValue = a();

console.log(
  'global: a() has returned.  Control has been given back to the global scope.'
);
console.log('global: Printing aValue:', aValue);
console.log('global: Script finished.');

// Output:
// global: Calling a() from the global scope and setting its value to aValue.
// a(): Inside of a().
// a(): Calling b() and setting its return value to bValue.
// b(): Inside of b().
// b(): Calling c() and setting its return value to cValue.
// c(): Inside of c().
// c(): Returning c's value.
// b(): c() has returned.  Control has been given back to b()
// b(): Printing cValue: It's me, c.
// b(): Returning B's value.
// a(): b() has returned.  Control has been given back to a().
// a(): Printing bValue:  It's me, b.
// a(): Returning a's value.
// global: a() has returned.  Control has been given back to the global scope.
// global: Printing aValue: It's me, a.
// global: Script finished.
```